src/util/log/log.cpp

```diff
@@ -2,11 +2,29 @@
 
 #include "../util_env.h"
 
+#include <ctime>
+#include <regex>
+
 namespace dxvk {
   
   Logger::Logger(const std::string& file_name)
-  : m_minLevel(getMinLogLevel()),
-    m_fileStream(file_name) { }
+  : m_minLevel(getMinLogLevel())
+  {
+    std::string path = env::getEnvVar(L"DXVK_LOG_PATH");
+    std::string file = path;
+    if (!file.empty() && *file.rbegin() != '/')
+      file += '/';
+    std::regex ext(".exe$");
+    file += std::regex_replace(env::getExeName(),ext,"") + "_";
```

Comment:
Wouldn't you be better off with something like
```cpp
std::string exe = env::getExeName();
std::string ext = exe.substr(exe.find_last_of('.'));
exe = ext == ".exe" ? exe.substr(0, exe.find_last_of('.') - 1) : exe;
file += exe;
```
?

Reply:
I wanted to keep it simple, once per log file at startup is not such big deal for performances. Guess it would save a include though.